src/plots/smith/index.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+var getSubplotCalcData = require('../get_data').getSubplotCalcData;
+var counterRegex = require('../../lib').counterRegex;
+
+var createSmith = require('./smith');
+var constants = require('./constants');
+
+var attr = constants.attr;
+var name = constants.name;
+var counter = counterRegex(name);
+
+var attributes = {};
+attributes[attr] = {
+    valType: 'subplotid',
+    dflt: name,
+    editType: 'calc',
+    description: [
+        'Sets a reference between this trace\'s data coordinates and',
+        'a smith subplot.',
+        'If *smith* (the default value), the data refer to `layout.smith`.',
+        'If *smith2*, the data refer to `layout.smith2`, and so on.'
+    ].join(' ')
+};
+
+function plot(gd) {
+    var fullLayout = gd._fullLayout;
+    var calcData = gd.calcdata;
+    var subplotIds = fullLayout._subplots[name];
+
+    for(var i = 0; i < subplotIds.length; i++) {
+        var id = subplotIds[i];
+        var subplotCalcData = getSubplotCalcData(calcData, name, id);
+        var subplot = fullLayout[id]._subplot;
+
+        if(!subplot) {
+            subplot = createSmith(gd, id);
+            fullLayout[id]._subplot = subplot;
+        }
+
+        subplot.plot(subplotCalcData, fullLayout, gd._promises);
+    }
+}
+
+function clean(newFullData, newFullLayout, oldFullData, oldFullLayout) {
+    var oldIds = oldFullLayout._subplots[name] || [];
+    var hadGl = (oldFullLayout._has && oldFullLayout._has('gl'));
+    var hasGl = (newFullLayout._has && newFullLayout._has('gl'));
+    var mustCleanScene = hadGl && !hasGl;
+
```

Comment:
No need for this extra logic here to handles WebGL base traces e.g. `scattersmithgl` ?

Reply:
8f67a6b0f49ddec580b96ef5cc2f8735663a7fbf